### ISSUE TITLE
LLAMA Demo Code Crashes my system

### ISSUE THREAD

**Issue by renxida** [nod-ai/SHARK-Turbine#107]
Ubuntu 22.04, AMD Ryzen 7 7800x3d, Nvidia RTX 4070

Syslog:

https://gist.github.com/renxida/c7a82b79dcf3682bde29072b04bebc23

Known triggers:
examples/llama2_inference/llama2.ipynb
examples/llama2_inference/stateless_llama.py

**Comment by renxida** [comment]
eager_mlp runs successfully.aot_mlp runs successfully

**Comment by renxida** [comment]
error in `llama2.ipynb` seems to be triggered by 

```


#Define some dim sizes to make the model code readable
BATCH_SIZE = 1 #Note: only 1 supported currently.
HEADS = 32
HIDDEN_DIM = 128
MAX_STEP_SEQ = 4095
#This is a maximum size past key value tensor, representing the full context window of the model. 
#We do this to make it easier for the compiler to reason about memory.
global_pkv = torch.zeros(
    size=(HEADS * 2, BATCH_SIZE, HEADS, MAX_STEP_SEQ, HIDDEN_DIM),
    dtype=torch.float32,
)
seq_step = AbstractIndex
```

**Comment by renxida** [comment]
Offending line:

global_pkv = torch.zeros(
    size=(HEADS * 2, BATCH_SIZE, HEADS, MAX_STEP_SEQ, HIDDEN_DIM),
    dtype=torch.float32,
)

Suggested solution: check that memory size is sufficient and throw an error if not, before doing this.

**Comment by renxida** [comment]
Would also be good if we have a smaller LLM to test.

**Comment by dan-garvey** [comment]
Yeah, this is a good point. We will add a smaller llm and I'll add a warning in the readme and the llama example about memory requirements. (I think its on the order of >128GB

**Comment by vivekkhandelwal1** [comment]
Hi @renxida, is this issue fixed?
